synapse_syntax_wishlist.md

```diff
@@ -0,0 +1,168 @@
+SELECT 2, 'Steve Ballmer', 5 UNION ALL
+SELECT 3, 'Satyla Nadella', 7;
+```
+
+## 2) `MERGE`
+
+> [relevant Uservoice Idea](https://feedback.azure.com/forums/307516-azure-synapse-analytics/suggestions/13520394--in-preview-merge-statement-support)
+
+This is already in public preview, but would love know when it becomes `GA`. When it does, we can drop [`synapse__snapshot_merge_sql`](dbt/include/synapse/macros/materializations/snapshot/snapshot_merge.sql) macro with it's `UPDATE->INSERT` workdaround and rely on [the global project's implementation](https://github.com/fishtown-analytics/dbt/blob/1060035838650a30e86989cbf2693db7720ff002/core/dbt/include/global_project/macros/materializations/snapshot/snapshot_merge.sql#L7-L25)
+
+```sql
+{% macro synapse__snapshot_merge_sql(target, source, insert_cols) -%}
+      {%- set insert_cols_csv = insert_cols | join(', ') -%}
+      EXEC('
+           BEGIN TRANSACTION
+           update {{ target }}
+          set dbt_valid_to = TMP.dbt_valid_to
+          from {{ source }} TMP
+          where {{ target }}.dbt_scd_id = TMP.dbt_scd_id
+            and TMP.dbt_change_type = ''update''
+            and {{ target }}.dbt_valid_to is null;
+
+            insert into {{ target }} (
+                  {{ insert_cols_csv }}
+                  )
+            select {{ insert_cols_csv }}
+            from {{ source }} 
+            where dbt_change_type = ''insert'' ; 
+           COMMIT TRANSACTION;
+           ');
+{% endmacro %}
+```
+
+
+## 3) Drop ... if exists
+###  `DROP [TABLE/VIEW/SCHEMA/INDEX] ... IF EXISTS`
+
+> [relevant Uservoice idea](https://feedback.azure.com/forums/307516-azure-synapse-analytics/suggestions/40068358-temporary-table-get-colunms-informations) ("suggested" 2016, "started" 2018 "planned" 2019)
+
+This one is a lower priority because IMHO, using the first statement enables all the products including SQL Server <2016. Though, the simplicity of the ssecond statement is alluring.
+#### Azure Synapse & SQL Server <2016
+```sql
+-- 
+if object_id ('dbo.clippy','V') is not null
+  begin
+  DROP VIEW dbo.clippy
+  end
+```
+#### Azure SQL & SQL Server >=201
+```sql
+DROP VIEW dbo.clippy IF EXISTS
+```
+## 4) Accessing data from blob or an external table
+### `OPENROWSET()` vs `CREATE EXTERNAL TABLE()` vs `COPY INTO`
+> [relevant Uservoice idea](https://feedback.azure.com/forums/307516-azure-synapse-analytics/suggestions/42118774-openrowset-for-dedicated-pools)
+
+These API dicrepancies are painfully confusing in that it's a grab bag of nouns. The asks are:
+- can there be common patterns for creating external tables across these TSQL products?
+- can Synapse support `RDBMS` External tables and Azure SQL support `HADOOP` External Tables?
+
+The impact on the dbt-msft project is how it affects `dbt-external-tables`, which currently supports rows 1 & 4 of the table below.
+
+There are huge UX gains left on the table with such disparate syntax across products. We'll add more specific samples later
```

Comment:
@chaerinlee1 dropping a note that as you finish the `COPY INTO` scripts for the getting started tutorials, that it would be great if you could use some of those example how "diverse" the syntax is